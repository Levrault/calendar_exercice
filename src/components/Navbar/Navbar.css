.navbar {
  background: #fff;
}

.list {
  @mixin horizontalList;
  border-bottom: 1px solid #f1f1f1;
  justify-content: space-between;
  padding: $space-m $space-l;
}

.sublist {
  @mixin horizontalList;
  justify-content: space-between;
}

.item {
  align-items: center;
  display: flex;
  margin-right: $space-m;
}
