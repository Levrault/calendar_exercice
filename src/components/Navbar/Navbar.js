import React from 'react';
import styles from './Navbar.css';
import YearFormConnected from '../YearBoxForm/YearFormConnected';
import Total from '../Total/Total';

/**
 * Main navigation component
 * @returns {node}
 */
const Navbar = () => (
  <header className={styles.navbar}>
    <ul className={styles.list}>
      <li className={styles.item}>
        Calendar App exercice
      </li>
      <li className={styles.item}>
        <ul className={styles.sublist}>
          <li className={styles.item}>
            <YearFormConnected />
          </li>
          <li className={styles.item}>
            <Total value={0} />
          </li>
        </ul>
      </li>
    </ul>
  </header>
);

export default Navbar;
