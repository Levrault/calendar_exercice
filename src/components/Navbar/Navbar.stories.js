import React from 'react';
import { storiesOf } from '@storybook/react';
import { withMarkdownNotes } from '@storybook/addon-notes';
import Provider from '../../tools/storybook/Provider';
import Navbar from './Navbar';

storiesOf('Nav', module)
  .addDecorator(story => <Provider story={story()} />)
  .add('default', withMarkdownNotes(`
    # Navbar

    A simple navbar
    `)(() => (
    <Navbar />
  )));

