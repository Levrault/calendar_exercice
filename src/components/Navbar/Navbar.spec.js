import React from 'react';
import { Provider } from 'react-redux';
import getFakeStore from '../../tools/test/fakeStore';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

describe('Navbar component', () => {
  it('should render without problem', () => {
    const store = getFakeStore();
    const component = renderer.create(
      <Provider store={store}>
        <Navbar />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
