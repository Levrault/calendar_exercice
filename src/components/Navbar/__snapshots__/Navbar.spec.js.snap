// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render without problem 1`] = `
<header>
  <ul>
    <li>
      Calendar App exercice
    </li>
    <li>
      <ul>
        <li>
          <form>
            <span>
              <button
                className="undefined"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="material-icons"
                >
                  chevron_left
                </i>
              </button>
              <input
                readOnly={true}
                type="number"
                value={2018}
              />
              <button
                className=""
                onClick={[Function]}
                type="button"
              >
                <i
                  className="material-icons"
                >
                  chevron_right
                </i>
              </button>
            </span>
          </form>
        </li>
      </ul>
    </li>
  </ul>
</header>
`;
